msweep.c

```diff
@@ -414,16 +422,29 @@ int main(int argc, char **argv) {
 		getkey(&key);
 		switch (key.type) {
 		case KNUM:
-			if (key.num > 1) {
+			if (have_repeat_num) {
+				if (repeat >= (INT_MAX - key.num) / 10) {  // would overflow
+					prflush("\x07");
+					repeat = 1;
+					have_repeat_num = false;
+				} else {
+					repeat = 10 * repeat + key.num;
+				}
+			} else if (key.num >= 1) {
 				repeat = key.num;
+				have_repeat_num = true;
 			}
 			break;
-		case KARROW:
+		case KARROW: {
+			int m = bd->w > bd->h ? bd->w : bd->h;
```

Comment:
Is this really needed, I mean it would take away some function calls, but `shiftcursor` doesn't allow the cursor to go outbounds.

Maybe we should create a `shiftcursor` which takes a count, I feel like it gets too messy too quickly otherwise.

Reply:
What would you think of calling `shiftcursor` 2147483647 times? ;)